### PR TITLE
Bug fix: "Export-AzAutomationRunbook Adding extra '\' to file names" and "Get-AzAutomationDscCompilationJobOutput returns truncated summaries"

### PR DESCRIPTION
**Pull request by Ab0907**
## Description

1.When using the cmdlet "Export-AzAutomationRunbook" in a linux environment, it would add an extra '\' to the name of each exported file. This bug was due to the use of '\' when forming the directory path which caused issues in Linux. The hard coded backslashes were replaced by 'DirectorySeparatorChar'

Find more details about the issue here: https://github.com/Azure/azure-powershell/issues/11101

2.Get-AzAutomationDscCompilationJobOutput returns truncated summaries.This cmdlet returns objects of type Microsoft.Azure.Commands.Automation.Model.JobStream and 'Summary' Property of these objects contains a truncated string

Find more details about the issue here: https://github.com/Azure/azure-powershell/issues/12322

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Automation/Automation/Common/AutomationPSClient.cs [hunk 1]
```diff
@@ -2013,7 +2013,8 @@ private DirectoryInfo WriteRunbookToFile(string outputFolder, string runbookName
 
             var fileExtension = IsGraphRunbook(runbookType) ? Constants.SupportedFileExtensions.Graph : Constants.SupportedFileExtensions.PowerShellScript;
 
-            var outputFilePath = outputFolderFullPath + "\\" + runbookName + fileExtension;
+            char seperator = Path.DirectorySeparatorChar;
+            var outputFilePath = outputFolderFullPath + seperator + runbookName + fileExtension;
```
Comment (BethanyZhou) [inline src/Automation/Automation/Common/AutomationPSClient.cs:2017]:
```suggestion
            var outputFilePath = Path.Join(outputFolderFullPath, seperator, runbookName + fileExtension);
```

File: src/Automation/Automation/Common/AutomationPSClient.cs [hunk 2]
```diff
@@ -2013,7 +2013,8 @@ private DirectoryInfo WriteRunbookToFile(string outputFolder, string runbookName
 
             var fileExtension = IsGraphRunbook(runbookType) ? Constants.SupportedFileExtensions.Graph : Constants.SupportedFileExtensions.PowerShellScript;
 
-            var outputFilePath = outputFolderFullPath + "\\" + runbookName + fileExtension;
+            char seperator = Path.DirectorySeparatorChar;
```
Comment (BethanyZhou) [inline src/Automation/Automation/Common/AutomationPSClient.cs:2016]:
```suggestion
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Please add change log for Automation. BTW, are you from Microsoft? If yes, please join Azure org to avoid we identify you as external contributor.

**Review by jagpreetstamber — DISMISSED** [review]

**Comment by Ab0907** [comment]
> Please add change log for Automation. BTW, are you from Microsoft? If yes, please join Azure org to avoid we identify you as external contributor.

Yep, will join the same

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

File: src/Automation/Automation/Common/AutomationPSClient.cs [hunk 3]
```diff
@@ -2013,7 +2013,7 @@ private DirectoryInfo WriteRunbookToFile(string outputFolder, string runbookName
 
             var fileExtension = IsGraphRunbook(runbookType) ? Constants.SupportedFileExtensions.Graph : Constants.SupportedFileExtensions.PowerShellScript;
 
-            var outputFilePath = outputFolderFullPath + "\\" + runbookName + fileExtension;
+            var outputFilePath = Path.Join(outputFolderFullPath, runbookName + fileExtension);
```
Comment (BethanyZhou) [inline src/Automation/Automation/Common/AutomationPSClient.cs:2016]: Did you use system.io namespace in this file?

**Comment by isra-fel** [comment]
Hi @Ab0907 please fix the build error. Thanks.

File: src/Automation/Automation/Common/AutomationPSClient.cs (hunk 3 above)
Comment (Ab0907) [reply src/Automation/Automation/Common/AutomationPSClient.cs:2016]: I did use System.IO namespace in this file but seems like Path.Join is only available in .NET Core. It's not in .NET Framework, and thus not available through netstandard. Causing the Error :"Path" does not contain a defination for "Join". A solution could be adding Path.DirectorySeparatorChar directly or using Path.Combine

**Comment by isra-fel** [comment]
Use this link to join Azure org https://aka.ms/AzureGitHub

**Review by jagpreetstamber — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**